pkl/decode_struct.go

```diff
@@ -305,7 +305,14 @@ func getStructFields(typ reflect.Type) map[string]structField {
 		field := typ.Field(i)
 		// embedded
 		if field.Anonymous {
-			for k, v := range getStructFields(field.Type.Elem()) {
+			var fields map[string]structField
+			switch field.Type.Kind() {
+			case reflect.Ptr:
+				fields = getStructFields(field.Type.Elem())
+			case reflect.Struct:
+				fields = getStructFields(field.Type)
+			}
+			for k, v := range fields {
```

Comment:
Not _very_ passionate, but shouldn't the difference be the "smallest"?
```suggestion
			fieldType := field.Type
			for fieldType.Kind() == reflect.Ptr {
				fieldType = fieldType.Elem()
			}
			for k, v := range getStructFields(fieldType) {
```
(as a side-effect; this also traverses _nested_ pointers)